pkg/actions/testdata/lua/catalogexport_hive_partition_pager.lua

```diff
@@ -0,0 +1,69 @@
+local hive = require("lakefs/catalogexport/hive")
+
+-- helper function to slice table array
+function table.slice(tbl, first, last, step)
+    local sliced = {}
+
+    for i = first or 1, last or #tbl, step or 1 do
+        sliced[#sliced + 1] = tbl[i]
+    end
+
+    return sliced
+end
+
+-- lakefs mock package
+
+local lakefs = {
+    list_objects = function(repo_id, commit_id, next_offset, prefix, delimiter, page_size)
+        local fs = {
+            [action.repository_id] = {
+                [action.commit_id] = {{
+                    physical_address = "s3://bucket/a1/b1/b",
+                    path = "letters/a=1/b=1/b.csv"
+                }, {
+                    physical_address = "s3://bucket/a2/b2/a",
+                    path = "letters/a=2/b=2/a.csv"
+                }, {
+                    physical_address = "s3://bucket/a2/b2/b",
+                    path = "letters/a=2/b=2/b.csv"
+                }, {
+                    physical_address = "",
+                    path = "letters/a=2/b=3/_SUCCESS"
+                }, {
+                    physical_address = "s3://bucket/a2/b3/a",
+                    path = "letters/a=2/b=3/a.csv"
+                }, {
+                    physical_address = "s3://bucket/a3/b4/a",
+                    path = "letters/a=3/b=4/a.csv"
+                }, {
+                    physical_address = "s3://bucket/a3/b4/b",
+                    path = "letters/a=3/b=4/b.csv"
+                }}
+            }
+        }
+        local all_entries = fs[repo_id][commit_id]
+        if next_offset == "" then
+            next_offset = 1
+        end
+        local end_idx = next_offset + page_size
+        return 200, {
+            results = table.slice(all_entries, next_offset, end_idx),
+            pagination = {
+                has_more = end_idx < #all_entries,
+                next_offset = end_idx + 1
+            }
+        }
+    end
+}
+
+local page = 2
```

Comment:
```suggestion
local page_size = 2
```
Not sure why we're using a different name than the parameter.

Also: can we check with multiple `page_size` arguments?  I would gladly check _every page size_ from 1 to the entire size of the list, or even the entire size of the list plus 2.  Each should give the same output.

Otherwise some pager bugs are just too easy to miss.

Reply:
Both points done, thanks!